rpc_util_test.go

```diff
@@ -294,3 +298,89 @@ func BenchmarkGZIPCompressor512KiB(b *testing.B) {
 func BenchmarkGZIPCompressor1MiB(b *testing.B) {
 	bmCompressor(b, 1024*1024, NewGZIPCompressor())
 }
+
+// compressData compresses data using gzip and returns the compressed bytes.
+// It now accepts *testing.T to handle errors during compression.
+func compressData(t *testing.T, data []byte) []byte {
+	var buf bytes.Buffer
+	gz := gzip.NewWriter(&buf)
+	if _, err := gz.Write(data); err != nil {
+		t.Fatalf("compressData() failed to write data: %v", err)
+	}
+
+	if err := gz.Close(); err != nil {
+		t.Fatalf("compressData() failed to close gzip writer: %v", err)
+	}
+	return buf.Bytes()
+}
+
+// compressInput compresses input data and returns a BufferSlice.
+func compressInput(input []byte) mem.BufferSlice {
+	compressedData := compressData(nil, input)
+	return mem.BufferSlice{mem.NewBuffer(&compressedData, nil)}
+}
```

Comment:
Please combine both these functions. Splitting them doesn't seem to provide any benefit in re-usability or abstraction, but slightly hinders readability by making readers scroll.